Composer/packages/server/src/controllers/project.ts

```diff
@@ -307,19 +307,43 @@ async function exportProject(req: Request, res: Response) {
   });
 }
 
-async function publishLuis(req: Request, res: Response) {
+async function setQnASettings(req: Request, res: Response) {
   const projectId = req.params.projectId;
   const user = await PluginLoader.getUserFromRequest(req);
 
   const currentProject = await BotProjectService.getProjectById(projectId, user);
   if (currentProject !== undefined) {
     try {
-      const luFiles = await currentProject.publishLuis(
-        req.body.luisConfig,
+      const qnaEndpointKey = await currentProject.updateQnaEndpointKey(req.body.subscriptionKey);
+      res.status(200).json(qnaEndpointKey);
+    } catch (error) {
+      res.status(400).json({
+        message: error instanceof Error ? error.message : error,
+      });
+    }
+  } else {
+    res.status(404).json({
+      message: 'No such bot project opened',
+    });
+  }
+}
+
+async function build(req: Request, res: Response) {
+  const projectId = req.params.projectId;
+  const user = await PluginLoader.getUserFromRequest(req);
+
+  const currentProject = await BotProjectService.getProjectById(projectId, user);
+  if (currentProject !== undefined) {
+    try {
+      const files = await currentProject.buildFiles(
```

Comment:
Before lubuild support an object type arguments, we still can implement our own object like
```
type BuildConfig {
  authoringKey: string;
  subscriptionKey: string;
  ...
}
```

When send to builder, deconstruct it or not depends on future lubuild implementations